unified-scheduler-logic/src/lib.rs

```diff
@@ -626,15 +626,21 @@ const_assert_eq!(mem::size_of::<UsageQueue>(), 8);
 pub struct SchedulingStateMachine {
     unblocked_task_queue: VecDeque<Task>,
     active_task_count: ShortCounter,
+    executing_task_count: ShortCounter,
+    max_executing_task_count: u32,
```

Comment:
why have a max executing task count? isn't it just the number of threads?

Reply:
i expanded documentation to answer that question: 55da9903a9f53954459f74842c31cc7d482eed1c